Xamarin.Forms.Platform.Android/Renderers/EditorRenderer.cs

```diff
@@ -291,12 +291,19 @@ void UpdateMaxLength()
 
 			currentFilters.Add(new InputFilterLengthFilter(Element.MaxLength));
 
-			EditText?.SetFilters(currentFilters.ToArray());
+			if (EditText == null)
+				return;
+
+			EditText.SetFilters(currentFilters.ToArray());
+			EditText.Text = TrimToMaxLength(EditText.Text);
+		}
 
-			var currentControlText = EditText?.Text;
+		string TrimToMaxLength(string currentText)
+		{
+			if (string.IsNullOrWhiteSpace(currentText) || currentText.Length <= Element.MaxLength)
```

Comment:
Why `OrWhiteSpace`? This would mean that one can put a lot of spaces in the editor?

Reply:
well observed, thanks!